src/controllers/addProduct.js

```diff
@@ -10,10 +10,23 @@ exports.post = (req, res) => {
   delete product.rating;
   delete product.comment;
 
-  // assign rating and comment to an object for passing to submitRating.js
+  // below I tried to isolate the problem and figured that this query might not do what it needs to do.
+  //The id always comes back as an empty array even with a productname that def. exists in the database
+  // let prodName = "Tofurky Feast";
+  // prodName = "'" + prodName + "'";
+  // console.log("prodname + quotes", prodName);
+  // getProductByName(prodName).then((id) => {
+  //   console.log(id);
+  // }
+  // )
 
   // Post the product & rating
-  postProduct(product).then(id => {
+  postProduct(product).then(() => {
+    let prodName = product.product_name;
+    prodName = "'" + prodName + "'";
+    console.log("prodname + quotes", prodName);
+    getProductByName(prodName);
+  }).then(id => {
```

Comment:
Nothing is being returned by this arrow function. .then() is only a function of a promise as such you would need to have this arrow function return a promise explicitly by creating it directly:
```
.then() => {
  return new Promise ((resolve, reject) => {
   ... rest of code
```
but there is no need as the `getProductsByName(prodName)` call is returning connection.query (which is this essentially http://vitaly-t.github.io/pg-promise/Database.html#query) and so is returning a promise. So you can just return that on line 28
` return getProductByName(prodName)` so that the promise returned by connection.query gets returned by that and it can be used in the `.then()` on line 29